src/catalog/catalog_entry/schema_catalog_entry.cpp

```diff
@@ -160,13 +160,16 @@ void SchemaCatalogEntry::DropEntry(ClientContext &context, DropInfo *info) {
 
 void SchemaCatalogEntry::AlterTable(ClientContext &context, AlterTableInfo *info) {
 	switch (info->alter_table_type) {
-	case AlterTableType::RENAME_TABLE: {
+	case AlterTableType::RENAME_TABLE:
+	case AlterTableType::RENAME_VIEW: {
 		auto &transaction = Transaction::GetTransaction(context);
 		auto entry = tables.GetEntry(transaction, info->table);
 		if (entry == nullptr) {
-			throw CatalogException("Table \"%s\" doesn't exist!", info->table);
+			throw CatalogException("Relation \"%s\" doesn't exist!", info->table);
 		}
-		assert(entry->type == CatalogType::TABLE_ENTRY);
+		CatalogType expected_type =
+		    info->alter_table_type == AlterTableType::RENAME_VIEW ? CatalogType::VIEW_ENTRY : CatalogType::TABLE_ENTRY;
+		assert(entry->type == expected_type);
```

Comment:
Is this checked before? Could you add a test where you use ALTER VIEW on a table, and ALTER TABLE on a view?

Reply:
Yep, we have both of those cases tested-- the first one is in `rename_view/test_rename_view_table.test` and the second one was already in place inside of `rename_table/test_rename_table_view.test`.